File: src/org/jetbrains/haskell/run/ConfigurationEditor.java
package org.jetbrains.haskell.run;

import com.intellij.openapi.module.Module;
import com.intellij.openapi.options.SettingsEditor;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

final class ConfigurationEditor extends SettingsEditor<HaskellRunConfiguration> {

    private final ProgramParamsPanel programParams;

    ConfigurationEditor(Module[] modules) {
        programParams = new ProgramParamsPanel(modules);
    }

    protected void applyEditorTo(HaskellRunConfiguration s) {
        programParams.applyTo(s);
    }

    protected void resetEditorFrom(HaskellRunConfiguration s) {
        programParams.reset(s);
    }

    @NotNull
    protected JComponent createEditor() {
        return programParams;
    }

    protected void disposeEditor() {
    }
}

File: plugin/src/org/jetbrains/haskell/run/haskell/ConfigurationEditor.kt
package org.jetbrains.haskell.run.haskell

import com.intellij.openapi.module.Module
import com.intellij.openapi.options.SettingsEditor
import javax.swing.*

class ConfigurationEditor(modules: Array<Module>) : SettingsEditor<CabalRunConfiguration>() {

    private val programParams: ProgramParamsPanel

    override fun applyEditorTo(s: CabalRunConfiguration?) {
        programParams.applyTo(s!!)
    }

    override fun resetEditorFrom(s: CabalRunConfiguration?) {
        programParams.reset(s!!)
    }

    override fun createEditor(): JComponent {
        return programParams
    }

    override fun disposeEditor() {
    }

    {
        programParams = ProgramParamsPanel(modules)
    }
}
